### PR TITLE
Switch the method names of `takeEffect()` and `doTakeEffect()`

### PR DESCRIPTION
**Pull request by keytsai**
#53

### CODE REVIEW

**Comment by gnr1118** [comment]
`public void ownedEstablishmentTakeEffect(Game game) {
       ownedEstablishment.forEach(establishment -> establishment.doTakeEffect(game, this));
}`

**Comment by keytsai** [comment]
> ```
> public void ownedEstablishmentTakeEffect(Game game) {
>     ownedEstablishment.forEach(establishment -> establishment.doTakeEffect(game, this));
> }
> ```
> 
> 這裡應該改成`establishment.takeEffect(game, this)` 其餘測試裡有呼叫到doTakeEffect方法的要一併修改

@gnr1118 漏改；已更新，並增加一個當前骰子點數若與所持卡牌所需骰子點數不同則不應發動效果的測試，以避免未注意到該情況

File: src/main/java/domain/card/establishment/Establishment.java [hunk 1]
```diff
@@ -11,26 +11,25 @@ public abstract class Establishment extends Card {
...
-    public Set<Integer> getDiceRollNeededToActivateEffect() {
-        return diceRollNeededToActivateEffect;
+    @Override
+    public void takeEffect(Game game, Player player) {
+        if (isDicePointToTakeEffect(game.getCurrentDicePoint())) {
+            doTakeEffect(game, player);
+        }
     }
 
+    public abstract void doTakeEffect(Game game, Player player);
```
[the remark this answers is not in the document]
Comment (keytsai) [reply src/main/java/domain/card/establishment/Establishment.java]: @gnr1118 已更新

[the remark this answers is not in the document]
Comment (gnr1118) [reply src/main/java/domain/card/establishment/Establishment.java]: 子類別要一併調整喔

[the remark this answers is not in the document]
Comment (keytsai) [reply src/main/java/domain/card/establishment/Establishment.java]: > 子類別要一併調整喔

@gnr1118 已更新

**Review by gnr1118 — APPROVED** [review]

**Pull request merged**